src/Utilities/CleverDicts.jl

```diff
@@ -62,22 +62,6 @@ mutable struct CleverDict{K,V,F<:Function,I<:Function} <: AbstractDict{K,V}
     set::BitSet
     vector::Vector{V}
     dict::OrderedCollections.OrderedDict{K,V}
-    function CleverDict{K,V}(n::Integer = 0) where {K,V}
-        set = BitSet()
-        sizehint!(set, n)
-        vec = Vector{K}(undef, n)
-        inverse_hash = x -> index_to_key(K, x)
```

Comment:
We need the type of `inverse_hash` to be easy to get to have the concrete type of the field of `VectorOfConstraints`. Here, you cannot get it with `typeof(some_function)` if the function is parametrized by `K`. Maybe we could do `typeof(Fix1(index_to_key, K))` if we use `Fix1` here.